shared/transaction/validator.ts

```diff
@@ -32,34 +32,45 @@ export class Validator {
 
     return (
       tx.vout.length === 2 &&
-      this.voutHas(tx.vout, new BigNumber(20000), ownerScript.stack) &&
-      this.voutHas(tx.vout, new BigNumber(10), createMasternodeOPCodes)
+      this.voutHasWithAmount(tx.vout, new BigNumber(20000), ownerScript.stack) &&
+      this.voutHasWithAmount(tx.vout, new BigNumber(10), createMasternodeOPCodes)
     );
   }
 
   private static resignMasternode(tx: CTransactionSegWit): boolean {
     return (
       tx.vout.length === 1 &&
-      this.voutHas(tx.vout, new BigNumber(0), [
+      this.voutHasWithAmount(tx.vout, new BigNumber(0), [
         OP_CODES.OP_RETURN,
         OP_CODES.OP_DEFI_TX_RESIGN_MASTER_NODE(undefined as unknown as ResignMasternode),
       ])
     );
   }
 
   private static sendFromLiq(tx: CTransactionSegWit, script: Script): boolean {
-    return tx.vout.length === 2 && this.voutContains(tx.vout, [[OP_CODES.OP_RETURN], script.stack]);
+    return (
+      tx.vout.length === 2 &&
+      !tx.vout[0].script.stack.map((code) => `${code.type}`).includes('OP_DEFI_TX') &&
```

Comment:
Is this a blockchain rule that the DeFi TX is always the first vout?

Reply:
yes. that is why there can only be one DEFI_TX inside one tx. As they need to be at index 0